### PR TITLE
Logout via Query Params

### PR DESCRIPTION
**Pull request by bwaidelich**
* Remove "cookieName" configuration and assume "&lt;serviceName&gt;-jwt" as
  cookie name
* Adjust `SetJwtCookieComponent` to handle logout (remove JWT cookie
  and trigger redirect) if `?logout=<serviceName>` query parameter is
  present

Closes: #2

### CODE REVIEW

**Comment by kitsunet** [comment]
That all looks pretty reasonable from my side and probably ready to merge, BUT why remove the cookieName configuration? Can't it be `$configuredCookieName . '-serviceName'` ?

File: Classes/Http/SetJwtCookieComponent.php [hunk 1]
```diff
@@ -78,13 +87,38 @@ private function isOpenIdConnectAuthentication(): bool
...
+
+    /**
+     * @param ComponentContext $componentContext
+     * @return void
+     */
+    private function handleLogout(ComponentContext $componentContext): void
+    {
+        $this->removeJwtCookie($componentContext);
+        $this->logger->info(sprintf('OpenID Connect Client: (%s) Logout requested (via query parameter) removing JWT cookie for service "%s".', get_class($this), $this->options['serviceName']));
+        $componentContext->replaceHttpResponse($componentContext->getHttpResponse()->withHeader('Location', (string)$componentContext->getHttpRequest()->getUri()->withQuery('')));
```
Comment (bwaidelich) [inline Classes/Http/SetJwtCookieComponent.php]: This part is a bit magic.. It redirects to the current URI without query params. If other params are required they would be missing. I think we have to revise this part, but without the redirect the account is already authenticated

Comment (kitsunet) [reply Classes/Http/SetJwtCookieComponent.php]: What about a defined "afterLogout" URL?

**Comment by bwaidelich** [comment]
Thanks for the review!
> why remove the cookieName configuration?
This package already has the notion of a "serviceName" to make it multi-tenant capable. Instead of duplicating the configuration everywhere we thought it would be better to "centralize" the oAuth tenants via the "serviceName".
But you're right, this is a breaking change and handling the logout in the SetCookieComponent is a bit awkward anyways..

I'll solve this in our application layer for now, until we found a better generic approach

**Comment by kdambekalns** [comment]
Gah, I guess conflict resolution via GH is a bad idea for all but the simples cases. Feel free to throw away https://github.com/flownative/flow-openidconnect-client/pull/3/commits/3ea22164a9a72661daddefdebbc6c63b857dfcaf again. Anyway, this touches code that has changed a bit in the meantime.

**Comment by bwaidelich** [comment]
The suggested solution was breaking and I'm not really happy with it.. Closing for now to remove this from my pending PRs

**Pull request closed without merging**